lib/msf/core/auxiliary/scanner.rb

```diff
@@ -119,6 +119,8 @@ def run
             if datastore['CHOST']
               @scan_errors << "The source IP (CHOST) value of #{datastore['CHOST']} was not usable"
             end
+          rescue Msf::Auxiliary::Failed => e
+            nmod.vprint_error("#{e}")
```

Comment:
Dispatching to `nmod.vprint_error` was the easiest / cleanest fix I found to properly handle the peer prefix. Calling it at this level ran into issues with the `AuthBrute` mixin used by the Jupyter example. Directly calling `vprint_error` would yield: `[-] 192.168.159.125-130:8888 - unreachable: Failed to fetch the Jupyter API version`.

Reply:
I agree `fail_with` should not abort the whole scan. However, I think this should be made optional. This could be done by overriding `fail_with` in this module and adding an optional boolean key word argument (e.g. `:abort`), which default to `false`. Maybe something along these lines:
```suggestion
            raise e if e.abort
            nmod.vprint_error("#{e}")
```

Another option would be to create a new kind of fail method (`fail_host_with`), which does not abort the all scan:
```ruby
  def fail_host_with(reason, msg = nil)
    vprint_error("#{reason.to_s}: #{msg}")
  end
```

Since `abort_on_exception` is `false` for these threads, unhandled exceptions will only be raised when `Thread#join` is called (line 151). Another option would be to add a "catch all" exception handler around the `join` and check if this should abort the scan or not:
```ruby
      begin
        @tl.first.join(1)
      rescue Exception => e
        raise e if e.respond_to?(:abort) && e.abort
        vprint_error("#{e}")
      end
```

Reply:
Cool, I think the first option with a new `kwarg` would be better. That would allow us to set a default that would be used by mixins instead of them aborting the entire module run. I'll take a look at implementing that.

Reply:
Fixed in 855aa3c5210ecf99899fae1e478664f84dbb29fd where I overrode `fail_with` to add the optional `abort` kwarg that when set to True would raise `Msf::Auxiliary::Failed` and abort the entire run. Alternatively, when `abort` is False, it'll raise the new `Msf::Auxiliary::Scanner::AttemptFailed` which will cause the current either batch or host to be failed but the next attempt will still be made.

Reply:
Awesome! this is even better! Thanks!